src/backend/renderer/pixman/mod.rs

```diff
@@ -0,0 +1,1208 @@
+use std::{
+    ops::Deref,
+    os::unix::io::{AsFd, BorrowedFd},
+    rc::Rc,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+use drm_fourcc::{DrmFormat, DrmFourcc, DrmModifier};
+use pixman::{Filter, FormatCode, Image, Operation, Repeat};
+use thiserror::Error;
+use tracing::warn;
+use wayland_server::{
+    protocol::{wl_buffer, wl_shm},
+    Weak,
+};
+
+use crate::{
+    backend::{
+        allocator::{dmabuf::Dmabuf, format::has_alpha, Buffer},
+        SwapBuffersError,
+    },
+    utils::{Buffer as BufferCoords, Physical, Rectangle, Scale, Size, Transform},
+    wayland::{compositor::SurfaceData, shm},
+};
+
+#[cfg(all(
+    feature = "wayland_frontend",
+    feature = "backend_egl",
+    feature = "use_system_lib"
+))]
+use super::ImportEgl;
+use super::{
+    sync::SyncPoint, Bind, DebugFlags, ExportMem, Frame, ImportDma, ImportDmaWl, ImportMem, ImportMemWl,
+    Offscreen, Renderer, Texture, TextureFilter, TextureMapping, Unbind,
+};
+
+bitflags::bitflags! {
+    struct DmaBufSyncFlags: std::ffi::c_ulonglong {
+        const READ = 1 << 0;
+        const WRITE = 2 << 0;
+        const START = 0 << 2;
+        const END = 1 << 2;
+    }
+}
+
+#[repr(C)]
+#[allow(non_camel_case_types)]
+struct dma_buf_sync {
+    flags: DmaBufSyncFlags,
+}
+
+struct DmabufIoctlSync {
+    sync: dma_buf_sync,
+}
+
+unsafe impl rustix::ioctl::Ioctl for DmabufIoctlSync {
+    type Output = ();
+
+    const OPCODE: rustix::ioctl::Opcode = rustix::ioctl::Opcode::write::<dma_buf_sync>(b'b', 0);
+
+    const IS_MUTATING: bool = true;
```

Comment:
If this is a "write" ioctl, and the kernel doesn't modify the data, this can be `false`.

But it looks like this is something the `rustix::ioctl::Setter` type could be used for, instead of manually implementing this trait.

```rust
type DmaBufSync = WriteOpcode<b'b', 0, dma_buf_sync>;
ioctl(fd, Setter<DmaBufSync, _>::new(dma_buf_sync { .. }));
```